modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java

```diff
@@ -0,0 +1,1144 @@
+				+ "from event_impression where hex(experiment_id) = '"
+				+ experiment.id.replace("-", "").toUpperCase()
+				+ "' and timestamp = '" + this.yesterday
+				+ "' group by bucket_label";
+		try (Statement statement = connection.createStatement();
+				ResultSet result = statement.executeQuery(queryGroupByBucket)) {
+			int userCount = 0;
+			int distinctUserCount = 0;
+			boolean atLeastOneRow = false;
+			while (result.next()) {
+				String bucket = result.getString("bucket_label");
+				userCount += result.getInt("user_count");
+				distinctUserCount += result.getInt("distinct_user_count");
+				atLeastOneRow = true;
+			}
+			if (!atLeastOneRow) {
+				fail("At least one row expected for query: "
+						+ queryGroupByBucket);
+			}
+
+			assertEquals(userCount, 3);
+			assertEquals(distinctUserCount, 3);
+		}
+	}
+
+	@Test(dependsOnMethods = { "t_PostImpressionTomorrowQA" })
+	public void t_CheckImpressionForYesterdayByBucketFromMySQLProdContext()
+			throws Exception {
+
+		String queryGroupByBucket = "select bucket_label, "
+				+ "count(user_id) as user_count, count(distinct user_id) as distinct_user_count  "
+				+ "from event_impression where hex(experiment_id) = '"
+				+ experiment.id.replace("-", "").toUpperCase()
+				+ "' and context = 'PROD' and timestamp = '" + this.yesterday
+				+ "' group by bucket_label";
+
+		try (Statement statement = connection.createStatement();
+				ResultSet result = statement.executeQuery(queryGroupByBucket)) {
+			int userCount = 0;
+			int distinctUserCount = 0;
+			boolean atLeastOneRow = false;
+			while (result.next()) {
+				String bucket = result.getString("bucket_label");
+				userCount += result.getInt("user_count");
+				distinctUserCount += result.getInt("distinct_user_count");
+				atLeastOneRow = true;
+			}
+			if (!atLeastOneRow) {
+				fail("At least one row expected for query: "
+						+ queryGroupByBucket);
+			}
+
+			assertEquals(userCount, 3);
+			assertEquals(distinctUserCount, 3);
+		}
+	}
+
+	@Test(dependsOnMethods = { "t_PostImpressionTomorrowQA" })
+	public void t_CheckImpressionForYesterdayByBucketFromMySQLQAContext()
+			throws Exception {
+
+		String queryGroupByBucket = "select bucket_label, "
+				+ "count(user_id) as user_count, count(distinct user_id) as distinct_user_count  "
+				+ "from event_impression where hex(experiment_id) = '"
+				+ experiment.id.replace("-", "").toUpperCase()
+				+ "' and context = 'QA' and timestamp = '" + this.yesterday
+				+ "' group by bucket_label";
+
+		try (Statement statement = connection.createStatement();
+				ResultSet result = statement.executeQuery(queryGroupByBucket)) {
+			if (result.next()) {
+				fail("Should have no rows for query: " + queryGroupByBucket);
+			}
+		}
+
+	}
+
+	@Test(dependsOnMethods = { "t_PostImpressionTomorrowQA" })
+	public void t_CheckImpressionForTomorrowByBucketFromMySQLProdContext()
+			throws Exception {
+
+		Statement statement = connection.createStatement();
```

Comment:
Here you can also use try-with-resource - this Statement and the ResultSet below are never closed.